test/widgets/dialog_checks.dart

```diff
@@ -15,22 +17,42 @@ Widget checkErrorDialog(WidgetTester tester, {
   required String expectedTitle,
   String? expectedMessage,
 }) {
-  final dialog = tester.widget<AlertDialog>(find.byType(AlertDialog));
-  tester.widget(find.descendant(matchRoot: true,
-    of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
-  if (expectedMessage != null) {
-    tester.widget(find.descendant(matchRoot: true,
-      of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
-  }
+  switch (defaultTargetPlatform) {
+    case TargetPlatform.android:
+    case TargetPlatform.fuchsia:
+    case TargetPlatform.linux:
+    case TargetPlatform.windows:
+      final dialog = tester.widget<AlertDialog>(find.bySubtype<AlertDialog>());
+      tester.widget(find.descendant(matchRoot: true,
+        of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
+      if (expectedMessage != null) {
+        tester.widget(find.descendant(matchRoot: true,
+          of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+      }
+
+      return tester.widget(find.descendant(of: find.byWidget(dialog),
+        matching: find.widgetWithText(TextButton, 'OK')));
+
+    case TargetPlatform.iOS:
+    case TargetPlatform.macOS:
+      final dialog = tester.widget<CupertinoAlertDialog>(find.byType(CupertinoAlertDialog));
+      tester.widget(find.descendant(matchRoot: true,
+        of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
+      if (expectedMessage != null) {
+        tester.widget(find.descendant(matchRoot: true,
+          of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+      }
 
-  return tester.widget(
-    find.descendant(of: find.byWidget(dialog),
-      matching: find.widgetWithText(TextButton, 'OK')));
+      return tester.widget(find.descendant(of: find.byWidget(dialog),
+        matching: find.widgetWithText(CupertinoDialogAction, 'OK')));
+  }
 }
 
-// TODO(#996) update this to check for per-platform flavors of alert dialog
-void checkNoErrorDialog(WidgetTester tester) {
-  check(find.byType(AlertDialog)).findsNothing();
```

Comment:
Ah, in https://github.com/zulip/zulip-flutter/pull/1017#discussion_r1978521358 I meant those two checks were redundant with each other. If there's no widget that's of a subtype of AlertDialog, there's certainly none that's of the exact type AlertDialog.